app/bundles/CampaignBundle/Entity/Event.php

```diff
@@ -635,6 +635,31 @@ public function getLog()
         return $this->log;
     }
 
+    /**
+     * Get log for a contact and a rotation.
+     *
+     * @param Contact $contact
+     * @param $rotation
+     * @return LeadEventLog|null
+     */
+    public function getLogByContactAndRotation(Contact $contact, $rotation)
+    {
+        $criteria = Criteria::create()
+            ->where(Criteria::expr()->eq('lead', $contact))
+            ->andWhere(Criteria::expr()->eq('rotation', $rotation))
+            ->orderBy(['id' => 'DESC'])
```

Comment:
This orderBy may be superfluous and decrease performance of the query.

Reply:
Fixed